src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -99,7 +99,8 @@ public MessageCacheItem getMessages(Iterator<Locale> fallbackLocalesIter) {
     	MessageCacheItem cacheItem = null;
     	if (cacheService.isContainComponent()) { // Item is in cache
     		cacheItem = cacheService.getCacheOfComponent();
-    		if (cacheItem.isExpired()) { // cacheItem has expired
+			// If the cacheItem is either expired or for a fallback locale
+    		if (cacheItem.isExpired() || !cacheItem.getLocale().equals(this.dto.getLocale())) {
```

Comment:
With this condition !cacheItem.getLocale().equals(this.dto.getLocale()),
cache default locale's content for requested locale may not take effect, because they never be equal.